src/1AC760.c

```diff
@@ -141,8 +145,652 @@ s32 calc_partner_test_enemy(void) {
+                    }
+
+                    if (dispatchEvent == EVENT_IMMUNE) {
+                        dispatchEvent = EVENT_FALL_TRIGGER;
+                    }
+
+                    tempBinary = TRUE;
+                }
+
+                if (gBattleStatus.flags1 & BS_FLAGS1_SP_EVT_ACTIVE) {
+                    if (battleStatus->currentAttackElement & DAMAGE_TYPE_JUMP && targetPart->eventFlags & ACTOR_EVENT_FLAG_FLIPABLE) {
+                        if (dispatchEvent == EVENT_HIT) {
+                            dispatchEvent = EVENT_FLIP_TRIGGER;
+                        }
+
+                        if (dispatchEvent == EVENT_IMMUNE) {
+                            dispatchEvent = EVENT_FLIP_TRIGGER;
+                        }
+
+                        if (!(target->flags & ACTOR_FLAG_400)) {
+                            tempBinary = TRUE;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    if (!(gBattleStatus.flags1 & BS_FLAGS1_SP_EVT_ACTIVE) && battleStatus->currentAttackElement & DAMAGE_TYPE_JUMP &&
+        targetPart->eventFlags & ACTOR_EVENT_FLAG_FLIPABLE) {
+        if (dispatchEvent == EVENT_HIT_COMBO) {
+            dispatchEvent = EVENT_FLIP_TRIGGER;
+        }
+
+        if (dispatchEvent == EVENT_UNKNOWN_TRIGGER) {
+            dispatchEvent = EVENT_FLIP_TRIGGER;
+        }
+
+        if (!(target->flags & ACTOR_FLAG_400)) {
+            tempBinary = TRUE;
+        }
+    }
+
+    if (gBattleStatus.flags1 & BS_FLAGS1_SP_EVT_ACTIVE) {
+        if (battleStatus->currentAttackElement & DAMAGE_TYPE_BLAST && targetPart->eventFlags & ACTOR_EVENT_FLAG_EXPLOSIVE) {
+            if (dispatchEvent == EVENT_DEATH) {
+                dispatchEvent = EVENT_EXPLODE_TRIGGER;
+            }
+
+            if (dispatchEvent == EVENT_IMMUNE) {
+                dispatchEvent = EVENT_EXPLODE_TRIGGER;
+            }
+
+            if (dispatchEvent == EVENT_HIT) {
+                dispatchEvent = EVENT_EXPLODE_TRIGGER;
+            }
+
+            tempBinary = TRUE;
+            isFireDamage = TRUE;
+        }
+
+        if (gBattleStatus.flags1 & BS_FLAGS1_SP_EVT_ACTIVE) {
+            if (battleStatus->currentAttackElement & (DAMAGE_TYPE_BLAST | DAMAGE_TYPE_FIRE)) {
+                if (dispatchEvent == EVENT_HIT) {
+                    dispatchEvent = EVENT_BURN_HIT;
+                }
+
+                if (dispatchEvent == EVENT_DEATH) {
+                    dispatchEvent = EVENT_BURN_DEATH;
+                }
+
+                isFireDamage = TRUE;
+            }
+
+            do {    // TODO remove this do while
+                if (gBattleStatus.flags1 & BS_FLAGS1_SP_EVT_ACTIVE && battleStatus->lastAttackDamage >= 0 && dispatchEvent != EVENT_DEATH &&
+                    dispatchEvent != EVENT_SPIN_SMASH_DEATH &&
+                    dispatchEvent != EVENT_EXPLODE_TRIGGER && !(targetPart->targetFlags & 4)) {
+                    #define INFLICT_STATUS(STATUS_TYPE) \
```

Comment:
Good idea 👍

Reply:
It wasn't mine I stole it from calc_item_damage_enemy